lib/PuppeteerSharp/Payload.cs

```diff
@@ -51,9 +51,14 @@ public string Hash
 
                 if (!normalizedUrl.StartsWith("data:", StringComparison.Ordinal))
                 {
-                    foreach (var item in Headers.Where(kv => kv.Key != "Accept" && kv.Key != "Referer" &&
-                                                               kv.Key != "X-DevTools-Emulate-Network-Conditions-Client-Id"))
+                    foreach (var item in Headers.OrderBy(kv => kv.Key))
                     {
+                        if (item.Key.Equals("Accept", StringComparison.OrdinalIgnoreCase)
+                            || item.Key.Equals("Referer", StringComparison.OrdinalIgnoreCase)
+                            || item.Key.Equals("X-DevTools-Emulate-Network-Conditions-Client-Id", StringComparison.OrdinalIgnoreCase))
+                        {
+                            continue;
+                        }
```

Comment:
Any reason we're changing this @Meir017?

Reply:
I had a failing test that was using lowercase
also - https://stackoverflow.com/questions/5258977/are-http-headers-case-sensitive#answer-5259004